### PR TITLE
Con 3317 reject job advert

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/Employer/Employer.Web/appsettings.Development.json [hunk 1]
```diff
@@ -29,11 +29,80 @@
...
   },
+  "ConnectionStrings": {
+    "Redis": "localhost:6379",
+    "MongoDb": "mongodb://dbadmin:changeme@localhost:27017/admin/?ssl=false&authMechanism=SCRAM-SHA-1",
+    "QueueStorage": "UseDevelopmentStorage=true",
+    "TableStorage": "UseDevelopmentStorage=true",
+    "CommunicationsStorage": "UseDevelopmentStorage=true"
+  },
+  "UseTableStorageQueryStore": false
+}*/
```
Comment (cofaulco) [inline src/Employer/Employer.Web/appsettings.Development.json:40]: can we remove commented out code?

File: src/Employer/UnitTests/Employer.Web/Orchestrators/VacancyPreviewOrchestratorTests.cs [hunk 2]
```diff
@@ -91,5 +91,58 @@ public VacancyPreviewOrchestratorTests()
...
+
+            _mockRecruitVacancyClient.Setup(x => x.GetVacancyAsync(vacancy.Id)).ReturnsAsync(vacancy);
+            _mockRecruitVacancyClient.Setup(x => x.GetEmployerDescriptionAsync(vacancy)).ReturnsAsync("employer description");
+            _mockRecruitVacancyClient.Setup(x => x.GetEmployerNameAsync(vacancy)).ReturnsAsync("employer name");
+            _mockRecruitVacancyClient.Setup(c => c.Validate(vacancy, It.IsAny<VacancyRuleSet>())).Returns(new EntityValidationResult());
+
+            var geocodeImageService = new Mock<IGeocodeImageService>();
+            var externalLinks = new Mock<IOptions<ExternalLinksConfiguration>>();
+            var mapper = new DisplayVacancyViewModelMapper(geocodeImageService.Object, externalLinks.Object, _mockRecruitVacancyClient.Object);
+            var shouldBeRejected = true;
```
Comment (cofaulco) [inline src/Employer/UnitTests/Employer.Web/Orchestrators/VacancyPreviewOrchestratorTests.cs:129]: no need for this local variable, this pattern used in the above test for multiple test cases

File: src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/RejectVacancyCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+{
+    public class RejectVacancyCommandHandlerTests
+    {
+        private readonly Mock<ILogger<RejectVacancyCommandHandler>> _mockLogger = new Mock<ILogger<RejectVacancyCommandHandler>>();
+        private readonly Mock<IVacancyRepository> _mockVacancyRepository = new Mock<IVacancyRepository>();
+        private readonly Mock<IMessaging> _mockMessaging = new Mock<IMessaging>();
+
+        [Theory]
+        [InlineData(VacancyStatus.Review)]        
+        public async Task WhenVacancyIsNotInValidState_ShouldSetVacancyReferred(VacancyStatus status)
```
Comment (cofaulco) [inline src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/RejectVacancyCommandHandlerTests.cs:24]: should this be WhenVacancy**IS**inValidState

**Review by cofaulco — COMMENTED** [review]
couple of minor comments

File: src/Employer/Employer.Web/appsettings.Development.json (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/Employer.Web/appsettings.Development.json]: done

File: src/Employer/UnitTests/Employer.Web/Orchestrators/VacancyPreviewOrchestratorTests.cs (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/Employer/UnitTests/Employer.Web/Orchestrators/VacancyPreviewOrchestratorTests.cs]: true, removed 👍

File: src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/RejectVacancyCommandHandlerTests.cs (hunk 3 above)
Comment (VasanthaKasirajan3008) [reply src/Shared/UnitTests/Vacancies.Client/Application/CommandHandlers/RejectVacancyCommandHandlerTests.cs]: changed the text.

**Review by cofaulco — APPROVED** [review]

**Pull request merged**